pages/project-version/middleware/index.js

```diff
@@ -110,8 +113,51 @@ const canComment = () => (req, res, next) => {
     .catch(next);
 };
 
+const getPreviousVersion = () => (req, res, next) => {
+
+  const versions = orderBy(req.project.versions.filter(pv => pv.id !== req.versionId),
+    v => moment(v.createdAt).format('YYYY-MM-DD'), ['desc']);
+
+  if (versions.length > 0) {
+    let prevVersion = versions.find(v => {
+      if (moment(req.version.createdAt).isAfter(v.createdAt)) {
+        return v;
+      }
+    });
+
+    req.api(`/establishments/${req.establishmentId}/projects/${req.projectId}/project-versions/${prevVersion.id}`)
+      .then(({ json: { data } }) => {
+        req.prevVersion = data;
+      })
+      .then(() => next())
+      .catch(next);
+  }
+
+};
+
+const getVersionChanges = () => (req, res, next) => {
+  if (req.prevVersion) {
+    const cvKeys = traverse(req.version.data, null, []);
+    const pvKeys = traverse(req.prevVersion.data, null, []);
+    const added = remove(cvKeys, k => !pvKeys.includes(k));
+    const removed = remove(pvKeys, k => !cvKeys.includes(k));
+    let changed = [];
+    cvKeys.forEach(k => {
+      let cvNode = getNode(req.version.data, k);
+      let pvNode = getNode(req.prevVersion.data, k);
+      if (!isEqual(cvNode, pvNode)) {
+        changed.push(k);
+      }
+    });
+    res.locals.static.changed = added.concat(removed).concat(changed);
```

Comment:
you'll also need to pass this to the client - have a look at `./js/index.js` - we pass the initial state when starting the app

Reply:
done